packages/vite/src/rsc/rscStudioHandlers.ts

```diff
@@ -0,0 +1,68 @@
+import http from 'node:http'
+import type { PassThrough } from 'node:stream'
+
+export const processRenderRscStream = async (
+  pipeable: PassThrough
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const chunks = [] as any
+
+    pipeable.on('data', (chunk: any) => {
+      chunks.push(chunk)
+    })
+
+    pipeable.on('end', () => {
+      const resultBuffer = Buffer.concat(chunks)
+      const resultString = resultBuffer.toString('utf-8') as string
+      resolve(resultString)
+    })
+
+    pipeable.on('error', (error) => {
+      reject(error)
+    })
+  })
+}
+
+export const postFlightToStudio = (payload: string) => {
+  const base64Payload = Buffer.from(payload).toString('base64')
```

Comment:
Am encoding this so it gets persisted as encoded text an not some unwieldily JSON with escape characters. Studio will decoded and handle the flight payload accordingly.